src/plugins/workspace/server/plugin.ts

```diff
@@ -115,12 +116,16 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
       if (path === '/') {
         const workspaceListResponse = await this.client?.list(
           { request, logger: this.logger },
-          { page: 1, perPage: 1 }
+          { page: 1, perPage: 2 }
         );
+        const basePath = core.http.basePath.serverBasePath;
+
         if (workspaceListResponse?.success && workspaceListResponse.result.total > 0) {
-          return toolkit.next();
+          return response.redirected({
+            headers: { location: `${basePath}/app/${WORKSPACE_NAVIGATION_APP_ID}` },
```

Comment:
We should have know which workspace to go in server side, we could just get the right url to go into the specific workspace.

Reply:
Sure, the code has been updated